### ISSUE TITLE
Random loss of settings

### ISSUE THREAD

**Issue by Simbiat** [Rem0o/FanControl.Releases#2589]
**Describe the bug**
Periodically (and seemingly randomly) Fan Control looses all settings: sensors linked to curves, mapping of fan controls to sensors, mapping of fan controls to curves, names of the fanes and sensors, calibration. Today (26.06.2024 it happened again). I think this has been happening at least for the past 4 versions, maybe more (current one for me is 194). No clear pattern, but possibly related to me using hibernation often.

**Is there a log.txt file next to FanControl.exe with recent date entries?**
[log.txt](https://github.com/user-attachments/files/15982447/log.txt)
Nothing relevant, at least not obviously relevant. At least nothing that for 26.06 when the issue occurred again.

**Relevant hardware specs and setup**
Motherboard TUF GAMING B650-PLUS with 2412 BIOS. No AOI, just 4 fans and Noctua NH-D15 with 2 fans. Logs also complain about my PSU HX1000i, but issue has been happening from before I got it (previous one was TUF Gaming 1000W Gold).

**Comment by Rem0o** [comment]
Your configuration(s) is/are saved in json files. Did you ever "save as" with a specific name?  Those files are located in the "Configurations" folder of the software.

**Comment by Simbiat** [comment]
No, no specific name, just default one `userConfig`. Only had 1 config, as well. Attaching current version of it for reference.
[userConfig.json](https://github.com/user-attachments/files/15991026/userConfig.json)

**Comment by Rem0o** [comment]
The userconfig.json you sent was restored right? It seems to have a correct config in it. I would recommend you save your config to a specific name and back it up somewhere.  

When the problem occur, do you get an error message saying it couldn't restore the config and that there are missing sensors?

**Comment by Simbiat** [comment]
I restored it manually, so yes, it should be correct. I do get errors about missing sensors after hibernation sometimes, but then refreshing restores everything. When this particular issue occurs there are no visible errors, just hear that fans are spinning faster, I open the app, and it looks as if it was reset, but all sensors are there.
Do you think saving config with custom name will prevent the error somehow?

**Comment by Rem0o** [comment]
The idea is just to get a backup/copy as a fallback.  TBH I don't know what's happening here.

**Comment by Simbiat** [comment]
Anything I can do to debug this on my end?

**Comment by Rem0o** [comment]
Do you have a "backup" folder in your configuration folder?  Does the date/time of those files match with the times you lost your config?

**Comment by Simbiat** [comment]
Time of the files themselves - don't think so, but time of the folder matches today's issue, so I would assume that the freshest file was moved/copied/created there today (attached for reference)
[V193_userConfig.json](https://github.com/user-attachments/files/15991732/V193_userConfig.json)
They all have different prefixes, which would suggest that this is related to updates, but... 194 update was installed yesterday and I had at least 1 reboot after that, so not sure if that's really related to the update, unless date change or something like that is also required. I do think, that previous instances of the issue occurred shortly after the updates.
![image](https://github.com/Rem0o/FanControl.Releases/assets/6022665/9ddfad14-3c04-479a-99c3-04fa42661f73)

**Comment by Rem0o** [comment]
The only thing that I could see is that:
LibreHardwareMonitor.Hardware.Psu.Corsair stuff in your logs.

That seems to throw errors upon specific conditions which fails the whole LHM library. When that occurs, it could mess up your config.  If you are not using any sensor from the corsair provider, I would uncheck the PSU section in sensor settings.

**Comment by Simbiat** [comment]
Unchecked it then got
![image](https://github.com/Rem0o/FanControl.Releases/assets/6022665/338fccfb-3cff-4f7e-b068-10807d9cde79)
After a couple of refreshes it was showing only PSU sensor, and I continued. Nothing new in the log, though. Will monitor how the app behaves now.

**Comment by Rem0o** [comment]
To be clear, your sensor settings should look like this:
![image](https://github.com/Rem0o/FanControl.Releases/assets/14118956/eff4ddeb-9922-42c8-b98e-f47410d44008)

**Comment by parkerbossier** [comment]
I believe I've encountered the same issue several times, including right now.

`new.json` is supposedly my last known-good state. It was last modified on 6/25, which I believe is when everything got reset, so it seems like it was automatically overridden by the app in a corrupt state. GPU sensors and controls were fine, but formerly hidden cards were showing, and the rest of the cards appeared as if I've never touched them before. My curve cards remained just fine. `new new.json` is what I recreated (this is now maybe the 6th time I've had to do this in a year or two). I just noticed that there's a "Backup" folder which has backups. `V193_new.json` seems to be a valid backup, and it's nice to know that I can use a backup going forward.

That said, I can confirm that the app has, on numerous occasions, seemingly lost most config data (as described above) and then forcefully silently overwrote my save.

[new new.json](https://github.com/user-attachments/files/16045362/new.new.json)
[new.json](https://github.com/user-attachments/files/16045363/new.json)
[V193_new.json](https://github.com/user-attachments/files/16045364/V193_new.json)

Let me know if you need additional info.
